File: Android/samples/demo/demo/src/main/java/com/didi/aoe/examples/demo/features/FeatureContents.java
package com.didi.aoe.examples.demo.features;

import com.didi.aoe.examples.demo.R;

import java.util.ArrayList;
import java.util.List;

import static com.didi.aoe.examples.demo.ShowcaseApp.getContext;

/**
 * @author noctis
 */
public class FeatureContents {
    private static final List<Feature> FEATURES = new ArrayList<>();

    static {

        FEATURES.add(new Feature(
                R.id.action_featuresFragment_to_mnistFeatureFragment,
                getContext().getString(R.string.title_mnist),
                getContext().getString(R.string.description_mnist),
                R.drawable.bg_mnist));

        FEATURES.add(new Feature(
                R.id.action_featuresFragment_to_squeezeFeartureFragment,
                getContext().getString(R.string.title_squeeze),
                getContext().getString(R.string.description_squeeze),
                R.drawable.bg_squeeze));
        FEATURES.add(new Feature(
                R.id.action_featuresFragment_to_recognizeFeartureFragment,
                getContext().getString(R.string.title_recognize),
                getContext().getString(R.string.description_recognize),
                R.drawable.bg_squeeze));

    }

    public static List<Feature> getFeatures() {
        return FEATURES;
    }
}

File: Android/examples/demo/src/main/java/com/didi/aoe/examples/demo/features/FeatureContents.kt
package com.didi.aoe.examples.demo.features

import com.didi.aoe.examples.demo.R
import com.didi.aoe.examples.demo.ShowcaseApp
import java.util.*

/**
 * @author noctis
 */
object FeatureContents {
    private val FEATURES = ArrayList<Feature>()

    val features: List<Feature>
        get() = FEATURES

    init {

        FEATURES.add(Feature(
                R.id.action_featuresFragment_to_mnistFeatureFragment,
                ShowcaseApp.context?.getString(R.string.title_mnist),
                ShowcaseApp.context?.getString(R.string.description_mnist)))

        FEATURES.add(Feature(
                R.id.action_featuresFragment_to_cameraFeatureFragment,
                ShowcaseApp.context?.getString(R.string.title_camera),
                ShowcaseApp.context?.getString(R.string.description_camera)))
    }
}
